packages/events/src/EventBoundary.ts

```diff
@@ -534,12 +534,21 @@ export class EventBoundary
 
     /**
      * Notify all the listeners to the event's `currentTarget`.
+     *
+     * If the `currentTarget` contains the property `on<type>`, then it is called here,
+     * simulating the behavior from version 6.x and prior.
      * @param e - The event passed to the target.
      * @param type
      */
     protected notifyTarget(e: FederatedEvent, type?: string): void
     {
         type = type ?? e.type;
+
+        // call the `on${type}` for the current target if it exists
+        if ((e.currentTarget as any)[`on${type}`])
+        {
+            (e.currentTarget as any)[`on${type}`](e);
+        }
```

Comment:
I think we can remove the `as any` cast here and replace it with something like this:
```suggestion
        e.currentTarget[`on${type}`]?.(e);
```

Reply:
I'd be fine with that personally, though ESLint isn't super happy about it. I get the following:

```
Element implicitly has an 'any' type because expression of type '`on${string}`' can't be used to index type 'FederatedEventTarget'
```

That was only avoided using the `as any` cast. I'm all for a more concise solution, though! Should that error be ignored, or should we think of another solution? (Thanks for your patience in this!)

Reply:
No problem! Thanks for taking the feedback. 

You could try cast  `on${string}` maybe as `as keyof IFederatedDisplayObject`

Reply:
Sorry for the delay, but there were additional challenges in getting this to be a bit more type safe.

What I ended up with was the following:

```javascript
       const displayObject = (e.currentTarget as unknown as IFederatedDisplayObject);

        (displayObject[`on${type}` as keyof IFederatedDisplayObject] as EventHandler)?.(e as FederatedPointerEvent);
```

I the cast to `unknown` wasn't ideal either, but it's not `any`, so there's that. It had to be split across 2 lines because of the line length limit (avoiding more warnings).

Hopefully this is more in line with best practices?